storage/named_query.go

```diff
@@ -61,6 +62,12 @@ var namedQueries = map[NamedQuery]string{
 	SELECT v.* FROM visibilities v
 	LEFT OUTER JOIN visibility_labels vl on v.id = vl.visibility_id
 	WHERE (vl.id IS NULL and v.service_plan_id in (:service_plan_ids))`,
```

Comment:
alternative query:
select * from visibilities v
left outer join visibility_labels l on v.id = l.visibility_id
where v.platform_id is null or l.id is null or (l.key=:key and l.val=:val)

Reply:
We see a big performance degradation when left outer join is used.
For example if we have 50k visibilities and 5mil visibility labels such join will create a big temporary set in which filtering using where is done after.

Our approach is to filter using where condition only. In our query some visibilities will be taken by platform criteria and for only for produced subset of visibilities a subquery will be performed.

Reply:
much more cumbersome but if you say big performance degradation I can't argue with that :)